### PR TITLE
Snap a roo

### PR DESCRIPTION
**Pull request by battlemidget**
More snappy improvements to get us running in classic mode with all the goodness

Testing:

```
snap download conjure-up
snap ack conjure-up*.assert
snap install conjure-up*.snap
conjure-up -d
```

### CODE REVIEW

File: conjureup/log.py [hunk 1]
```diff
@@ -24,10 +24,16 @@ def exception(self, msg):
...
-                                       backupCount=7)
+    try:
+        cmdslog = TimedRotatingFileHandler(logfile,
+                                           when='D',
+                                           interval=1,
+                                           backupCount=7)
+    except FileNotFoundError as e:
+        # SNAP: This failing
+        # May need to do something different for snappy
+        cmdslog = logging.FileHandler('deploy-done.log')
```
Comment (mikemccracken) [inline conjureup/log.py]: why 'deploy-done.log' - is this temporary to be fixed later?

File: conjureup/utils.py [hunk 2]
```diff
@@ -24,6 +25,15 @@
...
 
 
+def is_snap_package_installed(pkg):
+    # snapd is not idempotent so we need to query first
+    snap_query = 'http+unix://%2Frun%2Fsnapd.socket/v2/snaps/{}'.format(
+        pkg)
+    with requests_unixsocket.Session() as session:
+        if session.get(snap_query).ok:
+            return
+
```
Comment (mikemccracken) [inline conjureup/utils.py]: This function always returns None. as used, this is equivalent to always False - I'd suggest making it explicitly return True and False as appropriate.

File: snapcraft/script/bridge.start [hunk 3]
```diff
@@ -15,3 +13,6 @@ iptables -I INPUT -i conjureup0 -p tcp -m tcp --dport 53 -j ACCEPT
 iptables -I INPUT -i conjureup0 -p udp -m udp --dport 53 -j ACCEPT
 iptables -I INPUT -i conjureup0 -p tcp -m tcp --dport 67 -j ACCEPT
 iptables -I INPUT -i conjureup0 -p udp -m udp --dport 67 -j ACCEPT
+
+
+# security.idmap.isolated = True
```
Comment (mikemccracken) [inline snapcraft/script/bridge.start]: what's this doing here

File: conjureup/utils.py (hunk 2 above)
Comment (battlemidget) [inline conjureup/utils.py]: Nice catch I blame the snapcraft dev I stole this code from

File: snapcraft/script/bridge.start (hunk 3 above)
Comment (battlemidget) [inline snapcraft/script/bridge.start]: sorry note taking ill clean that up

File: conjureup/log.py (hunk 1 above)
Comment (battlemidget) [inline conjureup/log.py]: Yea still trying to figure out what's happening in the snapd environment

**Pull request merged**